protocols/mp-ecdsa/src/keystore.rs

```diff
@@ -0,0 +1,354 @@
+// Copyright 2022 Webb Technologies Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::convert::{From, TryInto};
+use std::sync::Arc;
+
+use parity_scale_codec::{Decode, Encode};
+use sp_application_crypto::{key_types::ACCOUNT, sr25519, RuntimeAppPublic, sp_core};
+use sp_keystore::{Keystore, KeystorePtr};
+
+use dkg_runtime_primitives::{
+    crypto::{Public, Signature},
+    KEY_TYPE,
+};
+use itertools::Itertools;
+use sc_keystore::LocalKeystore;
+use sp_application_crypto::sp_core::keccak_256;
+use crate::error;
+use crate::util::DebugLogger;
+
+/// A DKG specific keystore implemented as a `Newtype`. This is basically a
+/// wrapper around [`Keystore`] and allows to customize
+/// common cryptographic functionality.
+#[derive(Clone)]
+pub struct DKGKeystore(Option<KeystorePtr>, DebugLogger);
+
+impl DKGKeystore {
```

Comment:
If this is the MpEcdsa keystore we should name it such, I think getting more specific and semantically correct names would be beneficial here.